### PR TITLE
Identity: Unfurl UserID and Email in pkg/api to user identity.Requester

### PR DESCRIPTION
**Pull request by Jguer**
Identity: Unfurl UserID and Email in pkg/api to user identity.Requester

### CODE REVIEW

File: pkg/api/login.go [hunk 1]
```diff
@@ -329,7 +335,16 @@ func (hs *HTTPServer) RedirectResponseWithError(c *contextmodel.ReqContext, err
...
 	setCookie := true
 	if hs.Features.IsEnabled(featuremgmt.FlagIndividualCookiePreferences) {
-		prefsQuery := pref.GetPreferenceWithDefaultsQuery{UserID: c.UserID, OrgID: c.SignedInUser.GetOrgID(), Teams: c.Teams}
+		var userID int64
+		if c.SignedInUser != nil && !c.SignedInUser.IsNil() {
+			var errID error
+			userID, errID = identity.UserIdentifier(c.SignedInUser.GetNamespacedID())
+			if errID != nil {
+				hs.log.Error("failed to retrieve user ID", "error", errID)
+			}
```
Comment (IevaVasiljeva) [inline pkg/api/login.go:344]: nit: should we do `return hs.Cfg.AppSubURL + "/login"` from here? As we don't have the userID, so we shouldn't be able to successfully complete the preference query 🤔

File: pkg/api/render.go [hunk 2]
```diff
@@ -53,14 +54,19 @@ func (hs *HTTPServer) RenderToPng(c *contextmodel.ReqContext) {
...
+
 	result, err := hs.RenderService.Render(c.Req.Context(), rendering.Opts{
 		TimeoutOpts: rendering.TimeoutOpts{
 			Timeout: time.Duration(timeout) * time.Second,
 		},
 		AuthOpts: rendering.AuthOpts{
 			OrgID:   c.SignedInUser.GetOrgID(),
-			UserID:  c.UserID,
-			OrgRole: c.OrgRole,
+			UserID:  userID,
```
Comment (IevaVasiljeva) [inline pkg/api/render.go:68]: question: is there a danger in setting the `userID` to 0 here in case we have failed to `GetNamespacedID`?

**Review by IevaVasiljeva — APPROVED** [review]
LGTM overall! Left one question and one suggestion.

File: pkg/api/render.go (hunk 2 above)
Comment (Jguer) [reply pkg/api/render.go:68]: Probably not, the rendering service has been hitting it with user ID 0 already.
Still the type of endpoints I think we should revise in the future (stop it from being called with Anon for example)

**Pull request merged**